packages/ra-ui-materialui/src/list/pagination/Pagination.tsx

```diff
@@ -147,10 +147,47 @@ export const Pagination: FC<PaginationProps> = memo(props => {
     );
 });
 
+type rowsPerPageOptions = {
+    label: string;
+    value: number;
+};
+type rowsPerPageOptionsTypes = number[] | rowsPerPageOptions[];
+
+const rowsPerPageOptionsValidator: React.Validator<rowsPerPageOptionsTypes> = (
```

Comment:
No need to provide a validator for proptypes, as support for proptypes is obsolete and will probably be removed in the next major version. Just keep the existing proptype (`PropTypes.arrayOf(PropTypes.number)`).